### PR TITLE
ZeroLog v2

### PR DESCRIPTION
**Pull request by ltrzesniewski**
I've been working on a v2 of ZeroLog, and I'm currently in that weird no-man's land where I'm not sure if I changed too much or not enough... I's clearly not finished, but I'd like to have your input @rbouallou @MendelMonteiro @rverdier @farnyser about the new design.

Here are the main changes:

# General

- ZeroLog now targets .NET 6 only. A netstandard2.0 target needs to be added, which would *only* allow logging (no appenders, no config, etc).

# Loggers

- `InfoFormat` (etc) methods have been removed, they are replaced by the new interpolated string handlers feature of C# 10 😍, which means the following code:
  ```C#
  _log.Info($"Hello {World()}!");
  ```
  is equivalent to:
  ```C#
  if (_log.IsInfoEnabled)
  {
      _log.Info().Append("Hello ").Append(World()).Append("!").Log();
  }
  ```
- The `ILogEvent` *interface* has been removed, it's replaced by the `LogMessage` *class*, in order to remove lots of interface calls.
- The `ILog` *interface* has been removed, it's replaced by the `Log` *class*, for the same reason. It made sense to have a `Log`/`LogMessage` pair of classes. There is now a single implementation instead of 3.
- Calling `LogManager.GetLogger` before the `LogManager` is initialized will return a `Log` object which will work as soon as the log manager is initialized (it previously returned a no-op object which could never work).
- `LogEventArgumentExhaustionStrategy` has been removed. We now always truncate (it didn't work if the buffer was exceeded previously anyway).

# Appenders

- I added a `FormattedLogMessage` class which represents a formatted message (the logic was in `LogManager` previously). The `IAppender` API has been changed to account of this.
- StringFormatter has been removed. The library now uses the built-in features of .NET 6 (`ISpanFormattable`) that let us format values into a `Span<char>`. That one required *a lot* of changes... 😰
- Appenders can now each have a different encoding (makes sense for file vs console for instance).
- The library now allocates a large buffer that will never be extended during formatting (max line size = 16k chars).

# TODO

 - [x] Validate the new design (this PR)
 - [x] Add a limited netstandard2.0 target (logging only)
 - [x] I'd like to remove the JSON config, and replace it with something in the code. Also remove the JSON.NET library (a logging library should have no dependencies if possible).
 - [x] Appender construction could be better handled (we now supply every parameter separately to the constructor, which makes adding new parameters a breaking change).
 - [ ] Optimization/benchmarking
 - [ ] Add more unit tests
 - [x] There are still a few TODO comments in the source code
 - [x] Truncation support is incomplete right now (it won't truncate value types if string capacity is exceeded)
 - [x] Add an interpolated string handler overload to handle `.Append($"...")`
 - [ ] `DateOnly`/`TimeOnly`
 - [x] ~Ensure a message is not enqueued multiple times~
 - [x] Strong name
 - [x] Update the readme file
 - [x] XML documentation
 - [x] Test helpers
 - [x] Abstract formatting (to support logfmt for instance)

### CODE REVIEW

**Comment by MendelMonteiro** [comment]
The design choices that I have seen so far seem fairly common sense and tend to simplify the code.  It's awesome that we no longer depend on StringFormatter and that we can use the BCL formatting methods.

File: src/ZeroLog.sln [hunk 1]
```diff
@@ -29,6 +29,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZeroLog.Analyzers", "ZeroLo
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZeroLog.Analyzers.Tests", "ZeroLog.Analyzers.Tests\ZeroLog.Analyzers.Tests.csproj", "{FC5C8744-6E2C-49B0-9B3B-6BE6F825DF6C}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZeroLog.Tests.ExternalAppender", "ZeroLog.Tests.ExternalAppender\ZeroLog.Tests.ExternalAppender.csproj", "{92721CF3-E8E7-420B-9779-DCDDA8A336D9}"
```
Comment (ltrzesniewski) [inline src/ZeroLog.sln:32]: Oops, I wanted to delete this project but I apparently only removed it from the solution. It was used to make sure an external appender could be instantiated from the JSON config (that I removed).

Do you plan to use it, or should it be deleted?

Comment (MendelMonteiro) [reply src/ZeroLog.sln:32]: The solution wasn't building without it...

Comment (ltrzesniewski) [reply src/ZeroLog.sln:32]: Oh? It compiled fine for me... I'll make sure to delete it properly then.

**Pull request merged**